root/static/scripts/edit/utility/dates.js

```diff
@@ -44,7 +45,9 @@ export const isDateValid = function (y, m, d) {
 };
 
 export const isYearFourDigits = function (y) {
-  return (y === null || y === '' || y.length === 4);
+  return (y === null || y === '' ||
+          y.length === 4 && !isNaN(y.charAt(0)) ||
```

Comment:
Sorry, I was wrong because confused that with `Number.isNaN`. Though I think the `RegExp` might be more clear still - right now this is relying on the fact that `'-'` can't be coerced into a number, which is a bit weird.